### PR TITLE
Translate verification results from C++ to Rust

### PR DESCRIPTION
**Pull request by NlightNFotis**
and add tests for covering that.

- [ ] Each commit message has a non-empty body, explaining why the change was made.
- [ ] Methods or procedures I have added are documented, following the guidelines provided in CODING_STANDARD.md.
- [ ] The feature or user visible behaviour I have added or modified has been documented in the User Guide in doc/cprover-manual/
- [ ] Regression or unit tests are included, or existing tests cover the modified code (in this case I have detailed which ones those are in the commit message).
- [ ] My commit message includes data points confirming performance improvements (if claimed).
- [ ] My PR is restricted to a single feature or bugfix.
- [ ] White-space or formatting changes outside the feature-related changed lines are in commits of their own.

### CODE REVIEW

File: src/libcprover-rust/src/c_api.cc [hunk 1]
```diff
@@ -59,3 +59,33 @@ std::vector<std::string> const &get_messages()
...
+  *description = verifier_result->get_property_description(property_id);
+  return *description;
+}
+
+prop_statust get_property_status(
+  const std::unique_ptr<verification_resultt> &verifier_result,
+  const std::string &property_id)
+{
+  return verifier_result->get_property_status(property_id);
+}
```
Comment (TGWDB) [inline src/libcprover-rust/src/c_api.cc:91]: Please document the behaviour for a string that is not an ID for any property in the verifier result.

File: src/libcprover-rust/src/lib.rs [hunk 2]
```diff
@@ -21,20 +54,31 @@ pub mod cprover_api {
...
+        fn get_verification_result(result: &UniquePtr<verification_resultt>) -> verifier_resultt;
+        fn get_property_ids(result: &UniquePtr<verification_resultt>) -> &CxxVector<CxxString>;
+        fn get_property_description<'a>(
+            result: &'a UniquePtr<verification_resultt>,
+            property_id: &CxxString,
+        ) -> &'a CxxString;
+        fn get_property_status(
+            result: &UniquePtr<verification_resultt>,
+            property_id: &CxxString,
+        ) -> prop_statust;
```
Comment (TGWDB) [inline src/libcprover-rust/src/lib.rs]: Here is where the doc should be (as discussed).

File: src/libcprover-rust/src/lib.rs [hunk 3]
```diff
@@ -221,4 +268,145 @@ mod tests {
...
+                        "Property status for property {:?} was {:?} - expected FAIL",
+                        existing_property_id, prop_status
+                    );
+                    Err(error_msg)
+                }
+            }
+        } else {
+            Err("Unable to produce results from the verification engine".to_string())
+        }
+    }
```
Comment (TGWDB) [inline src/libcprover-rust/src/lib.rs:496]: Add test for bad inputs (those documented).

Comment (NlightNFotis) [reply src/libcprover-rust/src/lib.rs:496]: Done.

**Review by TGWDB — APPROVED** [review]

**Pull request merged**